include/jemalloc/internal/witness_inlines.h

```diff
@@ -92,14 +96,28 @@ witness_assert_lockless(tsdn_t *tsdn) {
 	}
 	tsd = tsdn_tsd(tsdn);
 
+	d = 0;
 	witnesses = tsd_witnessesp_get(tsd);
 	w = ql_last(witnesses, link);
 	if (w != NULL) {
-		witness_lockless_error(witnesses);
+		ql_reverse_foreach(w, witnesses, link) {
+			if (w->rank < rank_inclusive) {
+				break;
+			}
+			d++;
```

Comment:
Do you think it's necessary to add some sanity check in qr_foreach etc? e.g. add a counter and assume it's smaller than uint32_max.